splice/ingest.py

```diff
@@ -135,100 +135,86 @@ def populate_image_uris(tile, assets):
-                        command_logger.info("NOOP: id:{0} already exists".format(f_tile_id))
-
-                    else:
-                        """
-                        Either f_tile_id was not provided or
-                        the id's provided differ
-                        """
-                        t["directoryId"] = db_tile_id
-                        new_tiles_list.append(remove_unserializable_data(t))
-                        command_logger.info("IGNORE: Tile already exists with id: {1}".format(f_tile_id, db_tile_id))
+                    for dt_name in ('start_dt', 'end_dt'):
+                        dt = time_limits[dt_name]
+                        if dt and dt.tzinfo:
+                            # capture the datetime as UTC, but without the Timezone info
+                            # check because input may be TZ-unaware
+                            time_limits[dt_name] = dt.astimezone(pytz.utc).replace(tzinfo=None)
+
+                frequency_caps = t.get("frequency_caps", {"daily": 0, "total": 0})
+
+                columns = dict(
+                    target_url=t["url"],
+                    bg_color=t["bgColor"],
+                    title=t["title"],
+                    typ=t["type"],
+                    image_uri=image_hash,
+                    enhanced_image_uri=enhanced_image_hash,
+                    locale=locale,
+                    frecent_sites=frecent_sites,
+                    time_limits=time_limits,
+                    frequency_caps=frequency_caps,
+                    adgroup_name=adgroup_name,
+                    explanation=explanation,
+                    check_inadjacency=check_inadjacency,
+                    channel_id=channel_id,
+                )
+
+                try:
+                    with session_scope(conn) as session:
+                        db_tile_id, ag_id = tile_exists(session, **columns)
+                        f_tile_id = t.get("directoryId")
+
+                        if db_tile_id is None or ag_id is None:
+                            # Will generate a new id if not found in db
+                            db_tile_id, ag_id = insert_tile(session, **columns)
+                            t["directoryId"] = db_tile_id
+                            log_msg = "INSERT: Creating id:{0}".format(db_tile_id)
+
+                        elif db_tile_id == f_tile_id:
+                            log_msg = "NOOP: id:{0} already exists".format(f_tile_id)
+
+                        else:
+                            # Either f_tile_id was not provided or the id's provided differ
+                            t["directoryId"] = db_tile_id
+                            log_msg = "IGNORE: Tile already exists with id: {1}".format(f_tile_id, db_tile_id)
 
                 except Exception as e:
-                    trans.rollback()
                     command_logger.error("ERROR: {0}\nError inserting {1}.  ".format(e, json.dumps(
```

Comment:
Can we bubble up the error so that the originating caller, e.g. the API handler can return an error to the end-user?